crates/red_knot_python_semantic/src/types/class.rs

```diff
@@ -994,6 +1001,13 @@ impl<'db> KnownClass {
                     "ellipsis"
                 }
             }
+            Self::NotImplementedType => {
+                // Exposed as `types.NotImplementedType` on Python >=3.10;
+                // backported as `builtins._NotImplementedType` by typeshed on Python <=3.9
+                // TODO: Currently, types.NotImplementedType is a type alias of `builtins._NotImplementedType`
+                // TypeAlias is not fully supported yet, so we use `_NotImplementedType` here for now
```

Comment:
Not sure we need any of this comment, as mentioned below; I think the simplest approach will be to always consider `builtins._NotImplementedType` the canonical definition.